File: Lib/Telerik/Examples/QuickStart/GridView/Filtering/CustomFilterRow/ExpandableGridViewFilteringRowInfo.vb
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.GridView.Filtering.CustomFilterRow
	Public Class ExpandableGridViewFilteringRowInfo
		Inherits GridViewFilteringRowInfo
		#Region "Fields"

'INSTANT VB NOTE: The variable childRows was renamed since Visual Basic does not allow class members with the same name:
		Private childRows_Renamed As GridViewChildRowCollection

		#End Region

		#Region "Initialization"

		Public Sub New(ByVal view As GridViewInfo)
			MyBase.New(view)
			Me.SuspendPropertyNotifications()
			Me.IsExpanded = True
			Me.ResumePropertyNotifications()

			Me.childRows_Renamed = New GridViewChildRowCollection()
			Dim list As New List(Of GridViewRowInfo)()
			list.Add(New GridViewFilteringChildRowInfo(view))
			Me.childRows_Renamed.Load(list)
		End Sub

		#End Region

		#Region "Properties"

		Public Overrides ReadOnly Property HasChildViews() As Boolean
			Get
				Return Me.IsExpanded
			End Get
		End Property

		Public Overrides ReadOnly Property ChildRows() As GridViewChildRowCollection
			Get
				Return Me.childRows_Renamed
			End Get
		End Property

		#End Region
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/GridView/Filtering/CustomFilterRow/ExpandableGridViewFilteringRowInfo.cs
using System.Collections.Generic;
using Telerik.WinControls.UI;

namespace Telerik.Examples.WinControls.GridView.Filtering.CustomFilterRow
{
    public class ExpandableGridViewFilteringRowInfo : GridViewFilteringRowInfo
    {
        #region Fields

        private GridViewChildRowCollection childRows;

        #endregion

        #region Initialization

        public ExpandableGridViewFilteringRowInfo(GridViewInfo view)
            : base(view)
        {
            this.SuspendPropertyNotifications();
            this.IsExpanded = true;
            this.ResumePropertyNotifications();

            this.childRows = new GridViewChildRowCollection();
            List<GridViewRowInfo> list = new List<GridViewRowInfo>();
            list.Add(new GridViewFilteringChildRowInfo(view));
            this.childRows.Load(list);
        }

        #endregion

        #region Properties

        public override bool HasChildViews
        {
            get { return this.IsExpanded; }
        }

        public override GridViewChildRowCollection ChildRows
        {
            get { return this.childRows; }
        }

        #endregion
    }
}
